singer/src/main/java/com/pinterest/singer/writer/partitioners/FixedPartitionsPartitioner.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright 2019 Pinterest, Inc.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.pinterest.singer.writer.partitioners;
+
+import com.pinterest.singer.writer.KafkaMessagePartitioner;
+
+import org.apache.kafka.common.PartitionInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+/**
+ * This partition is used to group a batch of messages into one of at most three random partitions
+ */
+public class FixedPartitionsPartitioner implements KafkaMessagePartitioner {
+  public static final String PARTITIONER_PARTITION_COUNT_KEY = "partitioner.partition_count";
+  private static final Logger logger = LoggerFactory.getLogger(FixedPartitionsPartitioner.class);
+
+  private final Map<Integer, List<Integer>> partitionMap = new HashMap<>();
+  private final Random random = new Random(System.currentTimeMillis());
+  private int partitionCount = 3;
+
+
+  @Override
+  public void configure(Map<String, String> partitionerConfig) {
+    KafkaMessagePartitioner.super.configure(partitionerConfig);
+    partitionCount = Integer.parseInt(partitionerConfig.getOrDefault(PARTITIONER_PARTITION_COUNT_KEY, "3"));
```

Comment:
What's the reasoning for picking `3` as the default? I was thinking maybe we can use `1` as default so it matches the current single partition partitioner. This way we can use this new partitioner as the default for fixed partitioning and deprecate single partition partitioner going forward.